api/src/test/java/org/pmiops/workbench/captcha/CaptchaVerificationServiceTest.java

```diff
@@ -0,0 +1,104 @@
+  final String testAllOfUsUrl = "https://all-of-us-workbench-test.appspot.com";
+  final String responseToken = "responseToken";
+
+  @MockBean private CloudStorageService cloudStorageService;
+  @MockBean private WorkbenchConfig configProvider;
+  @MockBean private CaptchaApi captchaApiProvider;
+
+  @MockBean private WorkbenchConfig.AdminConfig adminConfig;
+  @Autowired private CaptchaVerificationServiceImpl captchaVerificationService;
+
+  @TestConfiguration
+  @Import({CaptchaVerificationServiceImpl.class, CaptchaApi.class})
+  static class Configuration {}
+
+  @Before
+  @Override
+  public void setUp() throws IOException {
+    super.setUp();
+    Mockito.when(cloudStorageService.getCaptchaServerKey()).thenReturn("key");
+    mockCaptchaResponse("hostName", true);
+  }
+
+  private void mockCaptchaResponse(String hostName, boolean success) {
+    CaptchaVerificationResponse response = new CaptchaVerificationResponse();
+    response.setHostname(hostName);
+    response.setSuccess(true);
+    try {
+      Mockito.when(captchaApiProvider.verify("key", responseToken)).thenReturn(response);
+    } catch (ApiException ex) {
+
+    }
+  }
+
+  @Test
+  public void testCreateAccount_invalidHostName() {
+    try {
+      // This should return false since the host name can either be google test hostname or one of
+      // AllOfUs urls
+      captchaVerificationService.mockLoginUrl("hostname");
+      boolean captchaSuccess = captchaVerificationService.verifyCaptcha(responseToken);
+      assertThat(captchaSuccess).isFalse();
+    } catch (ApiException e) {
+      e.printStackTrace();
+      assertThat(false);
+    }
+  }
+
+  /**
+   * For any of the AllOfUs urls not using captcha test keys, the hostName should match exactly with
+   * that send by google Captcha server
+   */
+  @Test
+  public void testCreateAccount_nonCaptchaTestHosts() {
+    try {
+      mockCaptchaResponse("workbench.researchallofus.org", true);
+      captchaVerificationService.mockLoginUrl(prodAllOfUsUrl);
+      boolean captchaSuccess = captchaVerificationService.verifyCaptcha(responseToken);
+      assertThat(captchaSuccess).isTrue();
+    } catch (ApiException e) {
+      e.printStackTrace();
+      assertThat(false);
+    }
+  }
+
+  @Test
+  public void testCreateAccount_googleTestKey() {
+    try {
+      // AllOfUs prod url should not be using google test captcha keys
+      mockCaptchaResponse("testkey.google.com", true);
+      captchaVerificationService.mockLoginUrl(prodAllOfUsUrl);
+      boolean captchaSuccess = captchaVerificationService.verifyCaptcha(responseToken);
+      assertThat(captchaSuccess).isFalse();
+
+      // AllOfUs test url should be using google test captcha keys
+      captchaVerificationService.mockLoginUrl(testAllOfUsUrl);
+      captchaSuccess = captchaVerificationService.verifyCaptcha(responseToken);
+      assertThat(captchaSuccess).isTrue();
+    } catch (ApiException e) {
```

Comment:
I would just add `throws Exception` to the unit test method definition throughout and remove these catches, I don't think the approach here offers any benefit over that